pkg/manifest/manifest.go

```diff
@@ -99,22 +148,77 @@ func ManifestsFromFiles(files []string) ([]Manifest, error) {
 }
 
 // ParseManifests parses a YAML or JSON document that may contain one or more
-// kubernetes resources.
-func ParseManifests(r io.Reader) ([]Manifest, error) {
+// kubernetes resources. Parsing and structure related errors are returned in
+// the first error parameter and cause an immediate return. Duplicate resource
+// errors are returned in the error list. Processing continues after a
+// duplicate resource error allowing all duplicates to be identified in the
+// given input.
+func ParseManifests(r io.Reader) ([]Manifest, error, []error) {
+	theseIds := make(map[ResourceId]bool)
+	var dupErrors []error
 	d := yaml.NewYAMLOrJSONDecoder(r, 1024)
 	var manifests []Manifest
 	for {
 		m := Manifest{}
 		if err := d.Decode(&m); err != nil {
 			if err == io.EOF {
-				return manifests, nil
+				return manifests, nil, dupErrors
 			}
-			return manifests, errors.Wrapf(err, "error parsing")
+			return manifests, errors.Wrapf(err, "error parsing"), dupErrors
 		}
 		m.Raw = bytes.TrimSpace(m.Raw)
 		if len(m.Raw) == 0 || bytes.Equal(m.Raw, []byte("null")) {
 			continue
 		}
+		if resourceId, err := unmarshalResourceId(m.Raw); err != nil {
```

Comment:
We probably don't have to unmarshal again.   Doesn't the `Decode` a few lines up call `*Manifest.UnmarshalJSON(...)` indirectly?

Reply:
Duh. It does and I had forgotten it worked that way.